src/sagemaker/session.py

```diff
@@ -866,7 +870,7 @@ def train(  # noqa: C901
             tensorboard_output_config=tensorboard_output_config,
             enable_sagemaker_metrics=enable_sagemaker_metrics,
             profiler_rule_configs=profiler_rule_configs,
-            profiler_config=profiler_config,
+            profiler_config=inferred_profiler_config,
```

Comment:
Is the a reason we are not supporting this in update_training_job() function? Refer Line 1152.

Reply:
I believe update_training_job is out of scope, we need to support **disable_profiler**  for CreateTrainingJob, in Estimator and Session according to the customer requirements doc but can double check

Reply:
Discussed offline: We will incorporate this using the same config injection for CreateTrainingJob schema and update in next commit